api/session/auth_session.py

```diff
@@ -111,22 +125,33 @@ def create_session_and_redirect(cls, account_id: str, redirect_url: str):
                 samesite=Config.FSD_USER_TOKEN_COOKIE_SAMESITE,
                 httponly=Config.SESSION_COOKIE_HTTPONLY,
             )
+            current_app.logger.info(f"User logged in to account : {account.id}")
             return response
         except SessionCreateError as e:
             error_response(404, str(e))
 
     @classmethod
-    def create_session_details_with_token(cls, account_id: str):
+    def create_session_details_with_token(
+            cls,
+            account: "Account",
+            timeout_seconds: int = Config.FSD_SESSION_TIMEOUT_SECONDS
+    ):
         """
-        Creates a signed expiring session token for the given account_id
-        :param account_id: The account_id for the user to create a token for
+        Creates a signed expiring session token for the given account
+        :param account: The account object for the user to create a token for
+        :param timeout_seconds: The length of the token expiry (or timeout)
         :return: A dict including the signed session token
         """
         session_details = {
-            "accountId": account_id,
+            "accountId": account.id,
+            "name": account.full_name,
+            "azureAdSubjectId": account.azure_ad_subject_id,
+            "email": account.email,
+            "fullName": account.full_name,
+            "roles": account.roles,
             "iat": int(datetime.now().timestamp()),
             "exp": int(
-                datetime.now().timestamp() + Config.FSD_SESSION_TIMEOUT_SECONDS
+                datetime.now().timestamp() + timeout_seconds
```

Comment:
This still defaults to CommonConfig.FSD_SESSION_TIMEOUT_SECONDS, however I've moved it to an argument so that SSO and Magic Links (which both use this function) can have separate session timeout lengths if required.